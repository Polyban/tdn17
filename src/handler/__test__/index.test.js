import { introHandler, sessionEndedHandler } from '../'
import launchRequest from '../../__mock__/launch-request.json'
import sessionEndedRequest from '../../__mock__/session-ended-request.json'

describe('Request handler', () => {
  const mockSession = {}

  it('should run introHandler', () => {
    const response = introHandler(mockSession, launchRequest)
    expect(response).toMatchSnapshot()
  })

  it('should run sessionEndedHandler', () => {
    const response = sessionEndedHandler(mockSession, sessionEndedRequest)
    expect(response).toMatchSnapshot()
  })
})
