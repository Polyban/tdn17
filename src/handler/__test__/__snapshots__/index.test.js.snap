// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request handler should run introHandler 1`] = `
Object {
  "response": Object {
    "card": Object {
      "image": Object {
        "largeImageUrl": "https://s3-eu-west-1.amazonaws.com/tdn17/01-home.3x.png",
        "smallImageUrl": "https://s3-eu-west-1.amazonaws.com/tdn17/01-home.2x.png",
      },
      "text": "Sage Informationen f端r Allgemeine Informationen oder Programm f端r eine Programm端bersicht.",
      "title": "Willkommen zum 35. Tag der Niedersachsen im Jahr 2017",
      "type": "Standard",
    },
    "outputSpeech": Object {
      "ssml": "<speak><p>Willkommen zum 35. \\"Tag der Niedersachsen\\" im Jahr 2017. <break /> Ich habe allgemeine Informationen oder eine Programm端bersicht.</p><p>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</p></speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {},
  "version": "1.0",
}
`;

exports[`Request handler should run sessionEndedHandler 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak><p>Auf Wiedersehen.</p></speak>",
      "type": "SSML",
    },
    "shouldEndSession": true,
  },
  "sessionAttributes": Object {},
  "version": "1.0",
}
`;
